internal/report/source.go

```diff
@@ -236,11 +236,21 @@ func assemblyPerSourceLine(objSyms []*objSymbol, rs graph.Nodes, src string, obj
 	srcBase := filepath.Base(src)
 	anodes := annotateAssembly(insts, rs, o.base)
 	var lineno = 0
+	var prevline = 0
 	for _, an := range anodes {
 		if filepath.Base(an.file) == srcBase {
 			lineno = an.line
 		}
 		if lineno != 0 {
+			if lineno != prevline && len(assembly[lineno]) > 0 {
+				// Place a separator between
+				// non-contiguous instructions.
+				sep := assemblyInstruction{
```

Comment:
Could we add a startBlock bool to assemblyInstruction instead of reusing the instruction field?
Or perhaps have this function return a map to a list of lists of assembly instructions.